### PR TITLE
ENH: Improve markup plane widget appearance

### LINKED ISSUE #4930: Enhance markups plane widget appearance
It would be useful to display plane boundary in 3D view (similarly how it is already done in slice view, but maybe with a thin tube instead of just a line).

We would also need flags to control what to display: boundary, filling, or both. Maybe we don't need new MRML attribute but the existing EdgeVisibility and Representation properties could be used.

### PR DESCRIPTION
**Pull request by Sunderlandkyl**
Improves the appearance of plane widgets by adding a border around the plane and allowing control of plane fill/border opacity.

fixes #4930

image

### CODE REVIEW

File: Modules/Loadable/Markups/MRML/vtkMRMLMarkupsDisplayNode.cxx [hunk 1]
```diff
@@ -231,6 +243,10 @@ void vtkMRMLMarkupsDisplayNode::CopyContent(vtkMRMLNode* anode, bool deepCopy/*=
   vtkMRMLCopyFloatMacro(LineColorFadingSaturation);
   vtkMRMLCopyFloatMacro(LineColorFadingHueOffset);
   vtkMRMLCopyBooleanMacro(HandlesInteractive);
+  vtkMRMLCopyBooleanMacro(FillVisibility);
+  vtkMRMLCopyBooleanMacro(OutlineVisibility);
+  vtkMRMLCopyFloatMacro(FillOpacity);
+  vtkMRMLCopyFloatMacro(OutlineOpacity);
```
Comment (lassoan) [inline Modules/Loadable/Markups/MRML/vtkMRMLMarkupsDisplayNode.cxx:249]: Could you add it to PrintSelf method, too?

Comment (Sunderlandkyl) [reply Modules/Loadable/Markups/MRML/vtkMRMLMarkupsDisplayNode.cxx:249]: Done!

**Comment by lassoan** [comment]
This is awesome. How difficult would it be to allow translating/rotating the plane by interacting with the border?

**Comment by Sunderlandkyl** [comment]
It should be possible. What operations would you expect to be mapped to each action?

**Comment by lassoan** [comment]
Great!

Since there is currently no consistent interaction style across different widgets in Slicer, we can just choose a mapping that seems the most intuitive (and maybe that is used in similar applications).

It could work similarly to Transforms module interaction widget: left-click-and-drag rotates, shift-left-click-and-drag translates; with refinements, such as rotating along one axis (the clicked axis; or when clicking near a corner then rotate along the plane normal). But it could work similarly to the slice reformat plane widget, which translates the plane when you click-and-drag it. We could also translate when the user clicks inside the plane (maybe only if interaction is enabled?).

Maybe we could go with anything that you find intuitive and then refine it based on user feedback.

**Comment by finetjul** [comment]
Nice !

Btw, it would be great if the slice view "Reformat" widget get replaced by this Plane markup (no point visible, only Interaction axes widget, and no translucency)

**Pull request merged**